### PR TITLE
Add link generator for comment filters

### PR DESCRIPTION
**Pull request by trymebytes**
In this pull request a link generator function for comment filter was added. Also, the display of the comment filters have been improved as shown below;
Screenshot 2022-01-18 at 02 54 38

### CODE REVIEW

File: helpers-assets/templates/translation-discussion-comments.php [hunk 1]
```diff
@@ -5,21 +5,22 @@
...
-			$countLocaleComments = 0;
-			foreach ( $comments as $comment ) {
-				$comment_locale = get_comment_meta( $comment->comment_ID, 'locale', true );
-				if ( $locale_slug == $comment_locale ) {
-					$countLocaleComments++;
-				}
-			}
-			?>
-						
+		<?php if ( ! $locale_slug ) : ?>
```
Comment (akirk) [inline helpers-assets/templates/translation-discussion-comments.php:8]: Is this still correct? Wouldn't it make more sense to check for `if ( $languages )` already here and swap the if contents?

File: helpers/helper-translation-discussion.php [hunk 2]
```diff
@@ -190,12 +190,37 @@ function() {
...
 				'locale_slug'        => $this->data['locale_slug'],
 				'original_permalink' => $this->data['permalink'],
+				'languages'          => $this->generateCommentFilterLinks( $comments, $this->data['locale_slug'] ),
 			),
 			$this->assets_dir . 'templates'
 		);
 		return $output;
 	}
 
+	public function generateCommentFilterLinks( $comments, $locale_slug ) {
```
Comment (akirk) [inline helpers/helper-translation-discussion.php:200]: I think the convention for function names is snake_case.

File: helpers/helper-translation-discussion.php [hunk 3]
```diff
@@ -190,12 +190,37 @@ function() {
...
+				}
+			}
+		}
+		$gp_locale = GP_Locales::by_slug( $locale_slug );
+		$locale_name = ( $gp_locale ) ? $gp_locale->english_name : 'Current locale';
+
+		return array(
+			$locale_slug => $locale_name . ' (' . $no_of_locale_comments . ')',
+			'all'        => 'Other locales (' . $plus_sign . $no_of_english_comments . ')',
+		);
```
Comment (akirk) [inline helpers/helper-translation-discussion.php:221]: This looks like it works but there are a few details that are not quite right, like `$no_of_english_comments` isn't just English comments. An idea for an improved version:

```suggestion
		if ( ! $comments ) {
			// Return early for edge cases.
			return array();
		}

		// Go through all $comments and map their values to the callback of this function.
		$comment_locales = array_map(
			function( $comment ) {
				// Each comment is passed in here and we return the locale.
				return get_comment_meta( $comment->comment_ID, 'locale', true );
			},
			$comments
		);

		// Produce an array of the form $locale => count( entries of array with $locale ).
		$comment_count = array_count_values( $comment_locales );

		// Let's extract the locale count and ensure that it was set.
		$locale_count = isset( $comment_count[ $locale_slug ] ) ? $comment_count[ $locale_slug ] : 0;
		$other_count = array_sum( $comment_count ) - $locale_count;

		$gp_locale = GP_Locales::by_slug( $locale_slug );
		$locale_name = $gp_locale ? $gp_locale->english_name : 'Current locale';

		$comment_links = array(
			$locale_slug => $locale_name . ' (' . $locale_count . ')',
		);

		if ( $other_count ) {
			$comment_links['all'] = 'Other locales (+' . $other_count . ')';
		}

		return $comment_links;
```

With that data in that structure you could also return all languages like this:
Screenshot 2022-01-19 at 15 22 09

```
		return array_filter( array_map( function( $locale_slug, $count ) {
			$locale_name = 'English';
			if ( $locale_slug ) {
				$gp_locale = GP_Locales::by_slug( $locale_slug );
				if ( ! $gp_locale ) {
					return null;
				}
				$locale_name = $gp_locale->english_name;
			}
			return $locale_name . ' (' . $count . ')';
		}, array_keys( $comment_count ), array_values( $comment_count ) ) );
```

**Review by akirk — CHANGES_REQUESTED** [review]
Works but it could use one more iteration for better code structure.

File: helpers/helper-translation-discussion.php (hunk 2 above)
Comment (trymebytes) [reply helpers/helper-translation-discussion.php:200]: Oops!

File: helpers-assets/templates/translation-discussion-comments.php (hunk 1 above)
Comment (trymebytes) [reply helpers-assets/templates/translation-discussion-comments.php:8]: This is in place to check if user is on the English discussions page.

File: helpers/helper-translation-discussion.php (hunk 3 above)
Comment (trymebytes) [reply helpers/helper-translation-discussion.php]: Thanks for this! Regarding the last sample code block listing all the other locales, that may contradict the change proposed in this issue - https://github.com/Automattic/Orbit/issues/138 - however, we can agree to go with the most appropriate solution.

**Pull request closed without merging**